ReactiveObjC/RACSignal+Operations.m

```diff
@@ -846,22 +846,57 @@ - (RACSignal *)takeUntilReplacement:(RACSignal *)replacement {
 
 - (RACSignal *)switchToLatest {
   return [[RACSignal createSignal:^(id<RACSubscriber> subscriber) {
-    RACMulticastConnection *connection = [self publish];
+    __block BOOL outerCompleted = NO;
+    __block BOOL innerCompleted = NO;
+    __block NSUInteger innerSignalIndex = 0;
 
-    RACDisposable *subscriptionDisposable = [[connection.signal
-      flattenMap:^(RACSignal *x) {
-        NSCAssert(x == nil || [x isKindOfClass:RACSignal.class], @"-switchToLatest requires that the source signal (%@) send signals. Instead we got: %@", self, x);
+    RACSerialDisposable *innerDisposable = [[RACSerialDisposable alloc] init];
+    NSLock *lock = [[NSLock alloc] init];
+
+    RACDisposable *outerDisposable = [self subscribeNext:^(RACSignal *signal) {
+      NSCAssert(signal == nil || [signal isKindOfClass:RACSignal.class], @"-switchToLatest expects signals as values, instead we got: %@", signal);
+      signal = signal ?: [RACSignal empty];
```

Comment:
minor: This can be moved to before the assert, in order to save the nil check in the assert.

Reply:
Done.

Reply:
As @alexger mentioned, for performance you'd want to avoid `-isKindOfClass:` if `signal` is `nil`.